DbgProvider/internal/Util.cs

```diff
@@ -416,7 +416,7 @@ public static string GetGenericTypeName( Type t, bool includeNamespace )
 
         public static string FormatErrorCode( int i )
         {
-            return FormatErrorCode( unchecked( (uint) i ) );
+            return FormatErrorCode( unchecked( (uint) i ) ) + " - " + Marshal.GetExceptionForHR( i ).Message;
```

Comment:
>" - " + Marshal.GetExceptionForHR( i ).Message; [](start = 62, length = 47)

I like the concept of this change, but I don't think this implementation will work very well--the CLR's mapping of error codes to exceptions is only a very small slice of error codes, and it's also informed by `IErrorInfo`, which when grabbing errors from a dump are not at all related.

Unfortunately I don't know of a good alternative. So for now let's just skip this particular change.

Reply:
Alright, I took some drastic measures on this one... and read the documentation. It turns out there's an overload with a second parameter for an IErrorInfo, with `(IntPtr)(-1)` instructing it to not use any IErrorInfo at all.

Aside from that, even if the mapping isn't very complete, it does cover the ones I encounter most frequently and I think it still gives a pretty reasonable message for the remainder.

Reply:
Nice find on the -1 trick.

Even if I liked the messages from `Marshal.GetExceptionForHR`, I think I'd still want a function that just formatted the numeric error value for some situations. And I think I probably just deal with a different set of error codes than you, and always seeing "0xNNNNNNNN - Exception from HRESULT: 0xNNNNNNNN" will really annoy me. So I'd like to skip this particular change.

---
In reply to: [260593827](https://github.com/Microsoft/DbgShell/pull/75#discussion_r260593827) [](ancestors = 260593827)

Reply:
Alright, it's removed